convert2rhel/actions/report.py

```diff
@@ -0,0 +1,145 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+
+import logging
+import textwrap
+
+from convert2rhel import utils
+from convert2rhel.actions import _STATUS_HEADER, find_actions_of_severity, format_action_status_message
+from convert2rhel.logger import colorize
+
+
+logger = logging.getLogger(__name__)
+
+#: Map Status codes (from convert2rhel.actions.STATUS_CODE) to color name (from
+#: convert2rhel.logger.bcolor)
+
+_STATUS_TO_COLOR = {
+    # SUCCESS
+    0: "OKGREEN",
+    # WARNING
+    51: "WARNING",
+    # SKIP
+    101: "FAIL",
+    # OVERRIDABLE
+    152: "FAIL",
+    # ERROR
+    202: "FAIL",
+}
+
+
+def summary(results, include_all_reports=False, with_colors=True):
+    """Output a summary regarding the actions execution.
+
+    This summary is intended to be used to inform the user about the results
+    reported by the actions.
+
+    .. note:: Expected results format is as following
+        {
+            "$Action_id": {
+                "status": int,
+                "error_id": "$error_id",
+                "message": None or "$message"
+            },
+        }
+
+    .. important:: Cases where the summary will be used
+        * All action_id have results that are successes (best case possible)
+            * If everything is a success, we just output a different message
+                for the user.
+        * Some action_id have results that are not successes (warnings, errors...)
+            * For those cases, we only want to print whatever is higher than
+                STATUS_CODE['WARNING']
+            * If we print something, let's try to use the correct logger
+                instead of just relying on `info`
+            * If one of the status has no corresponding logger function, we
+                should use just `info`
```

Comment:
That part does not make sense anymore, as we output the report with `info` for any message.
```suggestion
```